clang-tools-extra/clang-tidy/cppcoreguidelines/AvoidConstOrRefDataMembersCheck.cpp

```diff
@@ -13,79 +13,92 @@
 using namespace clang::ast_matchers;
 
 namespace clang::tidy::cppcoreguidelines {
-namespace {
 
-AST_MATCHER(FieldDecl, isMemberOfLambda) {
-  return Node.getParent()->isLambda();
+static bool hasCopyConstructor(CXXRecordDecl const &Node) {
```

Comment:
You can probably use `CXXRecordDecl::defaultedCopyConstructorIsDeleted` as well, at least for an earlier exit. (+ move ctor)

Reply:
`hasSimpleCopyConstructor` includes the `defaultedCopyConstructorIsDeleted`. The first check is for template function which the flag is not be setted.